base_rest/models/rest_service_registration.py

```diff
@@ -128,6 +128,17 @@ def _apply_default_if_not_set(self, controller_class, routing, attr_name):
         if hasattr(controller_class, default_attr_name) and attr_name not in routing:
             routing[attr_name] = getattr(controller_class, default_attr_name)
 
+    def _apply_default_auth_if_not_set(self, controller_class, routing):
+        default_attr_name = "_default_auth"
+        default_auth = getattr(controller_class, default_attr_name, None)
+        if default_auth:
+            if "auth" in routing:
+                auth = routing["auth"]
+                if auth == "public_or_default":
+                    routing["auth"] = "public_or_" + default_auth
```

Comment:
and where is this auth method handler defined?
I'm not sure to understand the purpose here :thinking:

Reply:
@simahawk public_or_jwt auth method over here (https://github.com/OCA/server-auth/pull/294) with an explanation of the use case.

Reply:
I see. Can you add something to the README and maybe point to that example or add a demo route?

Reply:
Can we check if `"_auth_method_public_or_" + default_auth` exists on `ir.http` ? Because, for instance, it does not make sense for auth_api_key where the client must always pass an api key (but sometimes does not pass PARTNER_EMAIL - but that is not managed in auth_api_key).

Reply:
@sbidoul done